codingGuideLines/cpp.md

```diff
@@ -273,48 +272,41 @@ foo( )
 ```
 
 
-## 10. Type Definitions
+## 10. Variable Declarations
   - the type qualifier `const` is placed right hand of the type that shall be const
     - note: `constexpr` is *not* a type qualifier, but an *expression qualifier*, write it *left* of the type!
   - `&` (reference) and `*` (pointer) **must** be surrounded by **one** space
 ```C++
-int const * byte;  // pointer to const int value
-int const * const byte; // const pointer to const int value
-int * const byte; // const pointer to int value
-int * byte; // pointer to int value
-int* byte; // NOT ALLOWED by the coding guide lines (missing spaces around `*`)
+int const * foo;  // pointer to const int value
+int const * const foo; // const pointer to const int value
+int * const foo; // const pointer to int value
+int * foo; // pointer to int value
+int* foo; // NOT ALLOWED by the coding guide lines (missing spaces around `*`)
 
 constexpr int i = 5; // this is a constant expression with ints
 ```
-
-
-## 11. Type Instantiations
-  - if possible do not use the assignment operator
+  - if possible use direct initialization over the assignment syntax
 ```C++
-float foo = 1; // should be avoided
-float foo( 1 ); // should be preferred, if possible
+int foo( 1 ); // should be preferred, if possible
```

Comment:
this rule is not the best anyway currently, we might want to remove it. it's especially odd for primitive types

Reply:
I agree it might me too strict to always require direct initialization syntax no matter what type is. However, I feel it's not for me to decide and my commit just improved the formulation with explicitly mentioning direct initialization and removing an inaccuracy with assigment.

Reply:
yes, if we would change it then we discuss it in a separate issue/pull request